pkg/tests/ossm/smcp_install_test.go

```diff
@@ -15,219 +15,101 @@
-		msg, _ := util.Shell(`oc get -n %s smcp/%s -o wide`, meshNamespace, smcpName)
-		if !strings.Contains(msg, "ComponentsReady") {
-			log.Log.Error("SMCP not Ready")
-			t.Error("SMCP not Ready")
-		}
-		util.Shell(`oc get -n %s pods`, meshNamespace)
+	oc.WaitSMCPReady(t, meshNamespace, smcpName)
+	oc.ApplyString(t, meshNamespace, smmr)
+	t.LogStep("Check SMCP is Ready")
+	retry.UntilSuccess(t, func(t TestHelper) {
+		oc.WaitCondition(t, meshNamespace, "smcp", smcpName, "Ready")
 	})
+}
 
-	t.Run("smcp_test_uninstall_2.1", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-		log.Log.Info("Delete SMCP v2.1 in ", meshNamespace)
-		util.KubeDeleteContents(meshNamespace, smmr)
-		util.KubeDeleteContents(meshNamespace, util.RunTemplate(smcpV21_template, smcp))
-		time.Sleep(time.Duration(60) * time.Second)
+func uninstallSMCPVersion(t test.TestHelper, smcpTemplate string, vars interface{}) {
+	t.LogStep("Delete SMCP in namespace " + meshNamespace)
+	oc.DeleteFromString(t, meshNamespace, smmr)
+	oc.DeleteFromTemplate(t, meshNamespace, smcpTemplate, vars)
+	retry.UntilSuccess(t, func(t TestHelper) {
+		oc.AllResourcesDeleted(t,
+			meshNamespace,
+			assert.OutputContains("No resources found in",
+				"All resources deleted from namespace",
+				"Still waiting for resources to be deleted from namespace"))
 	})
+}
 
-	t.Run("smcp_test_upgrade_2.1_to_2.2", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Create SMCP v2.1 in namespace ", meshNamespace)
-		util.ShellMuteOutputError(`oc new-project %s`, meshNamespace)
-		util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV21_template, smcp))
-		util.KubeApplyContents(meshNamespace, smmr)
-
-		// patch SMCP identity if it's on a ROSA cluster
-		if env.Getenv("ROSA", "false") == "true" {
-			util.Shell(`oc patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}'`, meshNamespace, smcpName)
-		}
-		log.Log.Info("Waiting for mesh installation to complete")
-		util.Shell(`oc wait --for condition=Ready -n %s smcp/%s --timeout 300s`, meshNamespace, smcpName)
-
-		log.Log.Info("Verify SMCP status and pods")
-		util.Shell(`oc get -n %s smcp/%s -o wide`, meshNamespace, smcpName)
-		util.Shell(`oc get -n %s pods`, meshNamespace)
-
-		log.Log.Info("Upgrade SMCP to v2.2")
-		util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV22_template, smcp))
-
-		// patch SMCP identity if it's on a ROSA cluster
-		if env.Getenv("ROSA", "false") == "true" {
-			util.Shell(`oc patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}'`, meshNamespace, smcpName)
+func TestSMCPInstall(t *testing.T) {
+	NewTest(t).Id("A1").Groups(Smoke, Full, InterOp, ARM).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		t.Cleanup(func() {
+			oc.DeleteNamespace(t, meshNamespace)
+			SetupNamespacesAndControlPlane()
+		})
+		vars := vars{
+			Name:      smcpName,
+			Namespace: meshNamespace,
```

Comment:
When executing templates, you can use a simple `map[string]string` instead of declaring a `struct`.

Reply:
Ok, going to change this